### ISSUE TITLE
Reproducibility of SAHP

### ISSUE THREAD

**Issue by ritvik06** [QiangAIResearcher/sahp_repo#6]
Hi,

Thanks a lot for your research and for providing the easily usable code. I tried to reproduce the results given in the paper for the StackOverflow dataset. I am able to reproduce the results for the macro-F1 score, in fact the results I got are slightly better than the reported results. But for RMSE, I don't get the intended results. What scale is the RMSE reported on in Table 4 in the paper? Also while preparing data before using the convert_real_data_syntheform.py file, do we need to normalise the timestamps or can we use the pkl's provided by you in the dataset link useable as is?

For more info, couple of hyper parameters I am using are as follows - 

Batch Size = 16
Hidden Size = 256
No. of attention heads = 8
number of layers = 4
learning rate = 1e-5
MC # of samples = 20

rest all parameters are default. Any help on this will be highly useful to me. Thanks!

**Comment by R-jr-star** [comment]
Me too, I tried the datasets "retweet" and "mimic", but I found the estimate_dt are all close to 0 so that the rmse is approximately equal to 1. Is there anyone having the same issue?

**Comment by gzagatti** [comment]
Thanks for making the source code available. I am also facing some reproducibility issues. @QiangAIResearcher it would be great to hear your thoughts on the findings from previous posts and the ones below.

I am running the model with defaults. Did you use different parameters? If so, which ones?

```sh
python3 main_func.py          \
  --epochs 1_000              \
  --batch 32                  \
  --lr 5e-5                   \
  --hidden 16                 \
  --atten-heads 8             \
  --pe "add"                  \
  --d-model 16                \
  --nLayers 4                 \
  --dropout 0.1               \
  --lambda-l2 3e-4            \
  --bias False                \
  --samples 10                \
  --early-stop-threshold 1e-2 \
  --train-ratio 0.8           \
  --dev-ratio 0.1             \
  --model "sahp"
```

I obtained the following results:

Dataset | LogLike | LogLike | LogLike | LogLike | RMSE | RMSE | F1 | F1
-- | -- | -- | -- | -- | -- | -- | -- | --
  | train | dev | test | test |   |   |   |  
  |   |   |   | paper |   | paper |   | paper
  | (nats) | (nats) | (nats) | (nats) |   |   |   |  
hawkes | 0.92 | 0.79 | 0.79 | 1.35 | 21.55 | 31.16 | 0.55 | 0.59
retweet | 6.08 | 5.99 | 5.95 | 4.56 | 1.00 | 1,055.05 | 0.54 | 0.54
so | 0.49 | 0.18 | 0.19 | 1.86 | 64.97 | 133.61 | 0.25 | 0.24
mimic | 2.40 | 2.49 | 2.69 | 0.52 | 5.18 | 3.89 | 0.27 | 0.37
meme | 6.95 | 6.85 | 7.00 |   | 1252.92 |   |   | 0.01


Some of the results were quite close, but there are some discrepancies:
- Hawkes (or simulated): better results in my case.
- Retweet: The RMSE is close to 1. I might be facing the same issue as @R-jr-star. F1-score is consistent.
- Stack Overflow: The paper uses only 1 data fold, while I have trained with all the folds. Is there any reason for using only 1-fold? However, I did obtain better results with regards to Log-likelihood and RMSE. F1-score is consistent
- Mimic: The paper uses only 1 data fold, while I have trained all the folds. In this case, I get worse results overall.


The paper does not use the `meme` dataset, although it is available in the shared data folder. There are also other datasets available in the same folder. Is there any reason for the selection you made in the paper?

**Comment by Wangzhenevergp** [comment]
> Thanks for making the source code available. I am also facing some reproducibility issues. @QiangAIResearcher it would be great to hear your thoughts on the findings from previous posts and the ones below.
> 
> I am running the model with defaults. Did you use different parameters? If so, which ones?
> 
> ```shell
> python3 main_func.py          \
>   --epochs 1_000              \
>   --batch 32                  \
>   --lr 5e-5                   \
>   --hidden 16                 \
>   --atten-heads 8             \
>   --pe "add"                  \
>   --d-model 16                \
>   --nLayers 4                 \
>   --dropout 0.1               \
>   --lambda-l2 3e-4            \
>   --bias False                \
>   --samples 10                \
>   --early-stop-threshold 1e-2 \
>   --train-ratio 0.8           \
>   --dev-ratio 0.1             \
>   --model "sahp"
> ```
> 
> I obtained the following results:
> 
> Dataset	LogLike	LogLike	LogLike	LogLike	RMSE	RMSE	F1	F1
>  	train	dev	test	test	 	 	 	 
>  	 	 	 	paper	 	paper	 	paper
>  	(nats)	(nats)	(nats)	(nats)	 	 	 	 
> hawkes	0.92	0.79	0.79	1.35	21.55	31.16	0.55	0.59
> retweet	6.08	5.99	5.95	4.56	1.00	1,055.05	0.54	0.54
> so	0.49	0.18	0.19	1.86	64.97	133.61	0.25	0.24
> mimic	2.40	2.49	2.69	0.52	5.18	3.89	0.27	0.37
> meme	6.95	6.85	7.00	 	1252.92	 	 	0.01
> Some of the results were quite close, but there are some discrepancies:
> 
> * Hawkes (or simulated): better results in my case.
> * Retweet: The RMSE is close to 1. I might be facing the same issue as @R-jr-star. F1-score is consistent.
> * Stack Overflow: The paper uses only 1 data fold, while I have trained with all the folds. Is there any reason for using only 1-fold? However, I did obtain better results with regards to Log-likelihood and RMSE. F1-score is consistent
> * Mimic: The paper uses only 1 data fold, while I have trained all the folds. In this case, I get worse results overall.
> 
> The paper does not use the dataset, although it is available in the shared data folder. There are also other datasets available in the same folder. Is there any reason for the selection you made in the paper?`meme`

How did you solve this problem?

**Comment by gzagatti** [comment]
Unfortunately, I could not resolve this issue. There remains a mismatch between the results in the paper and the results I obtained when running the source code in this repository.
